Allow path gizmos to be hidden

jimmcnulty41 opened the pull request:
Resolves #291 
Added an implementation for the layer service to path attributes.

`https://github.com/ild-games/duckling/issues/291`

Paths did not implement a function for the layer service. We now return a string "path-gizmos" for any entity that has a path attribute from a "getPathLayer" function which is bootstrapped into the entity-layer service

jimmcnulty41 committed:
Added a getLayer implementation for path attributes. This allows us to hide path gizmos.

tlein commented:
We should refactor how the entity-drawer.service uses the entity-layer.service so that it doesn't draw specific attributes based on the layers hidden instead of the entire entity.

As it is, if you add a drawable to an entity with a path and hide the layer the drawable is on it will still render.

jimmcnulty41 commented:
As in the drawable still renders?

tlein commented:
Yes, the drawable on the entity that is on the hidden layer will still render if that entity also has a path attribute.

jimmcnulty41 committed:
Use attributes, not entities, for the layers

* In enity-drawer, check visibility at the attribute level
* In entity-layer.service
  - Only update the layer service when it has actually changed on the store
  - Get Layer keys from individual attributes instead of the entity as a whole
  - Create isEntityAttributeVisible method to allow users to check for an individual entity
  - Change isEntityVisible to only return false when all attributes are invisible
  - Provide better defaults for the layerReducer

tlein commented on src/duckling/entitysystem/services/entity-layer.service.ts:
```diff
@@ -94,30 +83,49 @@ export class EntityLayerService extends BaseAttributeService<LayerGetter> {
     }
+
+    isEntityVisible(entity : Entity) : boolean {
+        // an entity is visible if any of its attributes are on visible layers
```
this comment isn't necessary

tlein commented on src/duckling/entitysystem/services/entity-layer.service.ts:
```diff
@@ -94,30 +83,49 @@ export class EntityLayerService extends BaseAttributeService<LayerGetter> {
+        return false;
+    }
+
+    isEntityAttributeVisible(entity: Entity, attributeKey: string){
```
add a space between the ){

jimmcnulty41 committed:
minor comment/formatting changes
